### ISSUE TITLE
JSON Array variation is broken on iOS

### ISSUE THREAD

**Issue by Titozzz** [launchdarkly/react-native-client-sdk#112]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `launchdarkly-react-native-client-sdk@5.1.1` for the project I'm working on.

Because LD SDK on iOS uses as? (see here: https://github.com/launchdarkly/ios-client-sdk/blob/v5/LaunchDarkly/LaunchDarkly/LDClient.swift#L473) to cast, the type is extra important. I wanted to use an empty array as my default value but wouldn't work. This fixes my issue, but might need some reviewing as I'm not a expert on the native side.

Do you want a PR ?

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/launchdarkly-react-native-client-sdk/ios/LaunchdarklyReactNativeClient.swift b/node_modules/launchdarkly-react-native-client-sdk/ios/LaunchdarklyReactNativeClient.swift
index 218e87d..7665ba0 100644
--- a/node_modules/launchdarkly-react-native-client-sdk/ios/LaunchdarklyReactNativeClient.swift
+++ b/node_modules/launchdarkly-react-native-client-sdk/ios/LaunchdarklyReactNativeClient.swift
@@ -144,8 +144,8 @@ class LaunchdarklyReactNativeClient: RCTEventEmitter {
         resolve(LDClient.get(environment: environment)!.variation(forKey: flagKey, defaultValue: defaultValue) as String)
     }
 
-    @objc func jsonVariationArray(_ flagKey: String, defaultValue: Array<RCTConvert>, environment: String, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) -> Void {
-        resolve(LDClient.get(environment: environment)!.variation(forKey: flagKey, defaultValue: defaultValue) as Array)
+    @objc func jsonVariationArray(_ flagKey: String, defaultValue: NSArray, environment: String, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) -> Void {
+        resolve(LDClient.get(environment: environment)!.variation(forKey: flagKey, defaultValue: defaultValue as Array) as Array)
     }
 
     @objc func jsonVariationObject(_ flagKey: String, defaultValue: NSDictionary, environment: String, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) -> Void {
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by louis-launchdarkly** [comment]
Hello @Titozzz, thank you for your contribution! I agree that we will need to review this before deciding what to do. Because the team in LaunchDarkly is trying to finish the work to support RN 0.65+, it will take some time for us before we can review this. I will give you an update when we have reviewed this, thank you for your patience.

_Filed Internally as 140977._

**Comment by Titozzz** [comment]
You mean the event emitter issues ? Appart from the warning if runs fine under 0.67. but removing those warning will be nice 👍

**Comment by gwhelanLD** [comment]
Hi @Titozzz,

Thanks for bringing this issue to our attention! It should be resolved in the new [6.0.0 release](https://github.com/launchdarkly/react-native-client-sdk/releases/tag/6.0.0).

Thanks,
@gwhelanLD
